### DISCUSSION TITLE
How can I migrate data from sentinel kv to kv-cluster?

### DISCUSSION THREAD

**Discussion by QQxiaoyuyu** [apache/kvrocks#2164] [Q&A] [answered]
I want to migrate data from sentinel kv to kv-cluster, and my sentinel kv version is 2.7, but my kv-cluster version is 2.8, Is there a problem with this.Or i need a kv-cluster  version is 2.7

**Comment by QQxiaoyuyu** [comment, marked as answer]
sentinel kv is redis-sentinel and master-slave

**Comment by caipengbo** [comment]
At the moment, I can think of 2 options: 
1. [kvrocks2redis](https://github.com/apache/kvrocks/tree/unstable/utils/kvrocks2redis):  It's single-threaded and can be very slow, and it won't sync to a cluster where the destination is multiple shards (a single shard works). If the incremental data is large, the temporary files generated during the migration can be very large. 
2. [RedisShake](https://github.com/tair-opensource/RedisShake): Select scan reader mode, which may also be slow for large data.

**Comment by QQxiaoyuyu** [comment]
redis-shake can't do it ,when i config over ,and running,now:
![image](https://github.com/apache/kvrocks/assets/34185590/79740244-b5fe-4404-8060-c61d95308652)

maybe kvrocks2redis can do it QAQ

**Comment by QQxiaoyuyu** [comment]
kvrocks2redis can not do it , I close .
